django/db/backends/base/schema.py

```diff
@@ -767,19 +726,70 @@ def _alter_field(self, model, old_field, new_field, old_type, new_type,
         # Drop the default if we need to
         # (Django usually does not use in-database defaults)
         if needs_database_default:
+            sql, params = self._alter_column_default_sql(model, old_field, new_field, True)
```

Comment:
Use `drop=True` rather than an arg for better readability.
Use a different var for `sql`here, maybe `changes_sql`? It's a little confusing to have the `sql` name reused in the next line.